registry/coder/modules/coder-login/main.tf

```diff
@@ -17,15 +17,14 @@ variable "agent_id" {
 data "coder_workspace" "me" {}
 data "coder_workspace_owner" "me" {}
 
-resource "coder_script" "coder-login" {
+resource "coder_env" "coder_session_token" {
   agent_id = var.agent_id
-  script = templatefile("${path.module}/run.sh", {
-    CODER_USER_TOKEN : data.coder_workspace_owner.me.session_token,
-    CODER_DEPLOYMENT_URL : data.coder_workspace.me.access_url
-  })
-  display_name       = "Coder Login"
-  icon               = "/icon/coder.svg"
-  run_on_start       = true
-  start_blocks_login = true
+  name     = "CODER_SESSION_TOKEN"
```

Comment:
The major change here is that this will cause the CLI to auth using the automatically generated session token directly. The previous behaviour would cause a new session token to be generated (`coder login --session-token=xxx` will cause a new token to be generated).

Users can still manually run `coder login` to maintain this behaviour, but highlighting it here.

Reply:
Thank you